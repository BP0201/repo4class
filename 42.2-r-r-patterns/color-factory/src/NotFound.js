import React from 'react';

import { Link } from 'react-router-dom'

const NotFound = () => {
    return (
        <div>
            <h1>Could not find the page you were looking for...</h1>
            <Link to="/colors">Click here</Link>
        </div>
    )
}

export default NotFound;