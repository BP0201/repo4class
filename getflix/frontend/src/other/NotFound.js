import React from 'react';

// simple component to display on nonexistent routes
const NotFound = () => {
    return (
        <div className='NotFound container my-5'>
            <div className='row justify-content-center'>
                <h3>404 Not Found</h3>
            </div>
            <div className='row justify-content-center'>
                <p>That route does not exist.</p>
            </div>
        </div>
    )
}

export default NotFound;