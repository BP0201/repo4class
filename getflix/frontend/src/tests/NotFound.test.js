import React from 'react';
import { render } from '@testing-library/react';
import NotFound from '../other/NotFound';

describe("NotFound", () => {
  it("matches snapshot", () => {
    const { asFragment, getByText } = render(<NotFound />);
    expect(asFragment()).toMatchSnapshot();
    expect(getByText("404 Not Found")).toBeInTheDocument();
  });
});