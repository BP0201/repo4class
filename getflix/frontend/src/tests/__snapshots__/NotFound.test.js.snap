// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="NotFound container my-5"
  >
    <div
      class="row justify-content-center"
    >
      <h3>
        404 Not Found
      </h3>
    </div>
    <div
      class="row justify-content-center"
    >
      <p>
        That route does not exist.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
